task1/task1.go

```diff
@@ -0,0 +1,73 @@
+package task1
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"strings"
+)
+
+type server struct {
+}
+
+type Query struct {
+	Key   interface{} `json:"key"`
+	Value interface{} `json:"value"`
+}
+
+type Headers struct {
+	Accept    []string `json:"Accept"`
+	UserAgent []string `json:"User-Agent"`
+}
+
+type Response struct {
+	Host       string  `json:"host"`
+	UserAgent  string  `json:"user_agent"`
+	RequestURI string  `json:"request_uri"`
+	Headers    Headers `json:"headers"`
+	Querys     []Query `json:"querys"`
+}
+
+func (s *server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	var qr []Query
+	for key, value := range r.URL.Query() {
+		var res interface{} = value
+		if len(value) == 1 {
+			res = value[0]
+		}
+		qr = append(qr, Query{
+			Key:   key,
+			Value: res,
+		})
+	}
+	resp := Response{
+		Host:       r.Host,
+		UserAgent:  r.UserAgent(),
+		RequestURI: r.RequestURI,
+		Headers: Headers{
+			Accept:    strings.Split(r.Header.Get("Accept"), ","),
+			UserAgent: []string{r.Header.Get("User-Agent")},
+		},
+		Querys: qr,
+	}
+	rJson, err := json.Marshal(resp)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		log.Panic(err)
```

Comment:
Please do not use panic here. It means that in case of error in process of handling request - whole server will be stopped.
Return an error and error code instead.

Reply:
oops, i fixed this 😅